syne_tune/optimizer/schedulers/searchers/bayesopt/contributed/estimator.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+import numpy as np
+
+from syne_tune.optimizer.schedulers.searchers.bayesopt.contributed.predictor import (
+    ContributedPredictor,
+    ContributedPredictorWrapper,
+)
+from syne_tune.optimizer.schedulers.searchers.bayesopt.datatypes.tuning_job_state import (
+    TuningJobState,
+)
+from syne_tune.optimizer.schedulers.searchers.bayesopt.models.estimator import (
+    Estimator,
+    transform_state_to_data,
+)
+from syne_tune.optimizer.schedulers.searchers.bayesopt.tuning_algorithms.base_classes import (
+    Predictor,
+)
+
+
+class ContributedEstimator:
+    """
+    Base class for the contributed Estimators
+    """
+
+    def fit(
+        self, X: np.ndarray, y: np.ndarray, update_params: bool
+    ) -> ContributedPredictor:
+        """
+        Implements :meth:`fit_from_state`, given transformed data.
+
+        :param X: Training data in ndarray of shape (n_samples, n_features)
+        :param y: Target values in ndarray of shape (n_samples,)
+        :param update_params: Should model (hyper)parameters be updated?
+        :return: Predictor, wrapping the posterior state
+        """
+        raise NotImplementedError()
+
+    @property
+    def normalize_targets(self) -> bool:
+        """
+        :return: Should targets in ``state`` be normalized before calling
+            :meth:`fit`?
+        """
+        return False
+
+
+class ContributedEstimatorWrapper(Estimator):
```

Comment:
I think this should also move to `models/contributed_estimator.py`, and why not inherit from `EstimatorFromTransformedData` ?

Reply:
This is just a rename of `EstimatorFromTransformedData`. I wanted to make estimator/predictor consistant with each other.